WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -232,7 +231,7 @@ extension PostEditor where Self: UIViewController {
         post = originalPost
         post.deleteRevision()
 
-        if shouldRemovePostOnDismiss {
+        if post.hasNeverAttemptedToUpload() && !post.isLocalRevision {
```

Comment:
I think `shouldRemovePostOnDismiss` was more descriptive of what was being checked, perhaps we can keep it as a computed property?

Reply:
I'm reluctant to have a property for something that's calculated only once and is a one liner &&. I also want to avoid having to lose the context of the method when you go check what `shouldRemovePostOnDismiss` really means.
But if you feel strongly about it I'll change it :D

Reply:
I agree with both (so helpful, I know!).

One middle ground option (that I often use when I want to provide more clarity) could be something along the lines of:

```swift
let shouldRemovePostOnDismiss =
        post.hasNeverAttemptedToUpload()
        && !post.isLocalRevision

if shouldRemovePostOnDismiss {
```

It doesn't really change the logic, but makes the code slightly more readable.

Reply:
I like it, changes made!